mono/utils/mono-time.c

```diff
@@ -152,7 +152,8 @@ mono_100ns_ticks (void)
 		timebase.denom *= 100; /* we return 100ns ticks */
 	}
 	return now * timebase.numer / timebase.denom;
-#elif defined(CLOCK_MONOTONIC)
+#elif defined(CLOCK_MONOTONIC) && !defined(_AIX)
+	/* !_AIX is defined because i 7.1 doesn't have clock_getres */
```

Comment:
Can't this be done using a configure check ?

Reply:
Not quite; because

1. we essentially do a cross-compile from AIX, which has functions i lacks; but the binaries are the same and work for both provided they stick to the same syscalls (and also because autotools itself is... basically broken on i)

2. i implements some functions in libc for the sake of linking compatibility, but these bomb out with SIGILL when invoked because the syscalls behind them are unimplemented (in this case, i 7.2 added them, but on i 7.1, nope)

Reply:
Can't we disable CLOCK_MONOTONIC in configure.in then ?

Reply:
That's a system-wide definition, and undefining it is an awfully big hammer for what's a small nail.

Reply:
Can't we check for clock_getres then?

Reply:
You could, but only if compiling on i; compiling from AIX wouldn't set such a thing, because AIX has it.